### PR TITLE
chore: upgrade Looker linter

### PR DESCRIPTION
**Pull request by bryans99**
Resulted in lots of changes but nothing significant. Main change is the header.

### CODE REVIEW

File: react/javascript/demo-core-sdk/src/demos/Dashboards.jsx [hunk 1]
```diff
@@ -72,21 +74,21 @@ export const Dashboards = () => {
   const columns = [
     {
       id: 'id',
+      size: 'small',
       title: 'ID',
       type: 'string',
```
Comment (jkaster) [inline react/javascript/demo-core-sdk/src/demos/Dashboards.jsx:79]: uh oh. is this that key sorting linter rule? Going to need to disable that if it gets reset in sdk-codegen

File: react/javascript/demo-extension-sdk/src/__mocks__/render_with_extension.js [hunk 2]
```diff
@@ -32,19 +34,25 @@ import { registerHostApi } from '@looker/extension-sdk'
...
   const extensionSDK = {
+    error: () => {
+      // noop
+    },
+    getContextData: () => {
+      // noop
+    },
+    localStorageGetItem: () => {
+      // noop
+    },
```
Comment (jkaster) [inline react/javascript/demo-extension-sdk/src/__mocks__/render_with_extension.js:45]: Did you insert these comments, or did the linter?

File: react/javascript/map-iframe/src/App.js [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
  */
 
 import React from 'react'
-import { MapIframe } from './MapIframe'
 import { ExtensionProvider2 } from '@looker/extension-sdk-react'
 import { hot } from 'react-hot-loader/root'
 import { Looker40SDK } from '@looker/sdk'
```
Comment (jkaster) [inline react/javascript/map-iframe/src/App.js:30]: not important for this PR but could `ILooker40SDK` be used here instead?

File: vanilla/counter-ts/src/index.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /*
  * The MIT License (MIT)
  *
- * Copyright (c) 2019 Looker Data Sciences, Inc.
+ * Copyright (c) 2022 Looker Data Sciences, Inc.
```
Comment (jkaster) [inline vanilla/counter-ts/src/index.ts:4]: Interesting this update left the `*` characters are the start of the line but most of the other updates didn't

File: scripts/create_uber_extension_project.ts [hunk 5]
```diff
@@ -250,7 +250,7 @@ function parseAndValidateArgs() {
   return { source, target, project, connection }
 }
 
-function isValidDirectory(name) {
+function isValidDirectory(name: string) {
```
Comment (jkaster) [inline scripts/create_uber_extension_project.ts:253]: Did you fix, or did the linter add `: string`?

File: react/typescript/kitchensink/src/components/CoreSDKFunctions/CoreSDKFunctions.tsx [hunk 6]
```diff
@@ -1,33 +1,35 @@
...
+
  */
 
 import isEqual from 'lodash/isEqual'
 import React, { useEffect, useState } from 'react'
 import { useLocation } from 'react-router-dom'
 import { Heading, Box, ButtonOutline, TextArea } from '@looker/components'
 import { getCoreSDK2 } from '@looker/extension-sdk-react'
-import { Looker40SDK } from '@looker/sdk'
+import type { Looker40SDK } from '@looker/sdk'
```
Comment (jkaster) [inline react/typescript/kitchensink/src/components/CoreSDKFunctions/CoreSDKFunctions.tsx:32]: Could this be `ILooker40SDK`?

**Review by jkaster — APPROVED** [review]
LGTM. I checked 163 of the files and the lint patterns were consistent. Minor observations along the way.

**Review by josephaxisa — COMMENTED** [review]
Looks ok to me. John raised some good questions though

File: react/javascript/demo-extension-sdk/src/__mocks__/render_with_extension.js (hunk 2 above)
Comment (bryans99) [reply react/javascript/demo-extension-sdk/src/__mocks__/render_with_extension.js:45]: I did those

File: vanilla/counter-ts/src/index.ts (hunk 4 above)
Comment (bryans99) [reply vanilla/counter-ts/src/index.ts:4]: Manually changed. Need to add linter rules to this. Another ticket though

File: scripts/create_uber_extension_project.ts (hunk 5 above)
Comment (bryans99) [reply scripts/create_uber_extension_project.ts:253]: I fixed it. These need to be linted as well. Another ticket

File: react/typescript/kitchensink/src/components/CoreSDKFunctions/CoreSDKFunctions.tsx (hunk 6 above)
Comment (bryans99) [reply react/typescript/kitchensink/src/components/CoreSDKFunctions/CoreSDKFunctions.tsx:32]: I'll look at it in the next set of dependabot fixes

**Pull request merged**